### PR TITLE
Read in Owner and ProjectName into Payload from .job.ad file

### PR DESCRIPTION
**Pull request by caseylargent**
Addresses #49

### CODE REVIEW

File: bin/stashcp [hunk 1]
```diff
@@ -97,6 +97,14 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
     payload['start2']=start2
     payload['end2']=end2
     
+    with open(".job.ad") as job_file:
```
Comment (bbockelm) [inline bin/stashcp]: - Might want to use a separate function for this; monitoring not directly related to this function
- Need to catch exceptions.
- Preferentially look at `$_CONDOR_JOB_AD` env var over the hardcoded filename.

File: bin/stashcp [hunk 2]
```diff
@@ -97,6 +97,14 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
...
+        for line in job_file.readLines():
+            if "AutoClusterAttrs" in line:
```
Comment (bbockelm) [inline bin/stashcp]: - You are going to have problems with any string that happens to contain these.  Do proper parsing of the line.
- ClassAd attribute names are not be case sensitive, but your parsing code is.

File: bin/stashcp [hunk 3]
```diff
@@ -97,6 +97,14 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
...
+                continue
+            if "Owner" in line:
+                payload["Owner"] = line[line.find("= ")]
+            if "ProjectName" in line:
+                payload["ProjectName"] = line[line.find("= ")]
```
Comment (bbockelm) [inline bin/stashcp]: This keeps unnecessary double-quotes, right?  Are there string escaping issues?

File: bin/stashcp (hunk 2 above)
Comment (caseylargent) [inline bin/stashcp]: So, is this kind of what your are looking for?
```Python
    with open(".job.ad") as job_file:
        for line in job_file.readLines():
            if line.lower().find("AutoClusterAttrs") == 0:
                continue
```
Or would you prefer going down the regex route

Comment (bbockelm) [inline bin/stashcp]: I think we'd be better off going down the regex route.

Comment (caseylargent) [inline bin/stashcp]: Alright, I think I have something that may be of use
```python
        for line in job_file.readLines():
            match = re.match('Owner|ProjectName', line, re.IGNORECASE)
            if match:
                payload[match.group()] = line
```
re.match only matches at the beginning of the string. Looking good?

File: bin/stashcp [hunk 4]
```diff
@@ -141,6 +143,19 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
...
 
 
+def parse_job_ad():
+    temp_list = {}
+    try:
+        with open(os.environ['_CONDOR_JOB_AD']) as job_file:
+            for line in job_file.readLines():
+                match = re.match('Owner|ProjectName', line, re.IGNORECASE)
+                if match:
+                    temp_list[match.group()] = line
```
Comment (bbockelm) [inline bin/stashcp]: I think this is the right approach.

However, I'm not sure if this quite works?  Did you test this?  Seems like it's going to put the full line in as the group.  It also still keeps quotes.

You probably want a regex like this:
```
r = re.compile('^\s*(Owner|ProjectName)\s=\s"(.*)"', re.IGNORECASE)
```

Seems that we need a small unit test for this function, given it's not 100% clear what the output is.

Comment (caseylargent) [inline bin/stashcp]: I tested it, but your regex is less prone to error. Sorry, not the best with regex haha. Where would the unit test be placed?

**Comment by djw8605** [comment]
@caseylargent all the builds are failing here: https://travis-ci.org/opensciencegrid/StashCache/jobs/211014419#L1406

File: bin/stashcp [hunk 5]
```diff
@@ -141,6 +143,19 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
         es_send(payload)
 
 
+def parse_job_ad():
+    temp_list = {}
+    try:
+        with open('.job.ad') as job_file:
```
Comment (djw8605) [inline bin/stashcp]: Why did you change this from reading in the environment?  I think reading in the environment is the best option.

Comment (caseylargent) [inline bin/stashcp]: Reading in from the environment was causing the build to fail beforehand, even though I had the exact name of the condor environment variable

**Comment by djw8605** [comment]
Could you do some logic, testing:

1. Check if environment variable is there
2. Check if .job.ad file is in the current directory.

If both of these tests fail, just don't add the username or project.

File: bin/stashcp [hunk 6]
```diff
@@ -141,6 +143,23 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
         es_send(payload)
 
 
+def parse_job_ad():
+    temp_list = {}
+    try:
+        if '_CONDOR_JOB_AD' in os.environ:
+            filename = os.environ['_CONDOR_JOB_AD']
+        else:
+            filename = "job.ad"
```
Comment (djw8605) [inline bin/stashcp]: The file name is `.job.ad` I believe.

File: bin/stashcp [hunk 7]
```diff
@@ -141,6 +143,23 @@ def doStashCpSingle(sourceFile, destination, cache, debug=False):
...
+        with open(filename) as job_file:
```
Comment (djw8605) [inline bin/stashcp]: Only attempt to open the file if the environment variable or the file exists.

Comment (djw8605) [inline bin/stashcp]: Check for the existence with `os.path.exists(filename)`
